src/entity/campaign/Campaign.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  Column,
+  Entity,
+  Index,
+  ManyToOne,
+  PrimaryColumn,
+  TableInheritance,
+} from 'typeorm';
+import { User } from '../user';
+
+export enum CampaignType {
+  Post = 'post',
+  Source = 'source',
+}
+
+export enum CampaignState {
+  Pending = 'pending',
+  Active = 'active',
+  Completed = 'completed',
+  Cancelled = 'cancelled',
+}
+
+export interface CampaignFlags {
+  budget: number;
+  spend: number;
+  impressions: number;
+  clicks: number;
+  users: number;
+}
+
+@Entity()
+@TableInheritance({ column: { type: 'varchar', name: 'type' } })
+export class Campaign {
+  @PrimaryColumn({ type: 'uuid', generated: 'uuid' })
+  id: string;
+
+  @Column({ type: 'text' })
+  referenceId: string;
+
+  @Column({ type: 'text' })
+  userId: string;
+
+  @ManyToOne('User', {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  user: Promise<User>;
+
+  @Column({ type: 'text' })
+  @Index('IDX_campaign_type')
+  type: CampaignType;
+
+  @Column({ default: () => 'now()' })
+  createdAt: Date;
```

Comment:
`CreateDateColumn`, can you also add UpdateDateColumn, also add sorting index on createdAt since you use it below

Reply:
Makes sense. I should probably add index on the state as it was also used for the ordering. Wdyt?

Reply:
Updated!

https://github.com/dailydotdev/daily-api/pull/2969/files#diff-4daa1d167c44768292e10f41da1e9bbc5fc967a2c50cfd16b38f50a47ed50375R22-R24